hedera-node/src/main/java/com/hedera/services/context/ServicesContext.java

```diff
@@ -598,11 +599,20 @@ public ServicesContext(
 		this.platform = platform;
 		this.state = state;
 		this.propertySources = propertySources;
+
+		updateWorkingState(state);
 	}
 
+	/**
+	 * Update the state and working state based on the provided service state
+	 *
+	 * @param state latest state from the services
+	 */
 	public void update(ServicesState state) {
 		this.state = state;
 
+		updateWorkingState(state);
```

Comment:
Very nice! If we used the same pattern to manage the queryable state, that would make the code much cleaner.